lib/galaxy/tools/data/__init__.py

```diff
@@ -786,11 +796,114 @@ def to_dict(self):
+
+    type_key = 'refgenie'
+
+    def __init__(self, config_element, tool_data_path, from_shed_config=False, filename=None, tool_data_path_files=None, other_config_dict=None):
+        super(RefgenieToolDataTable, self).__init__(config_element, tool_data_path, from_shed_config, filename, tool_data_path_files, other_config_dict=other_config_dict)
+        self.config_element = config_element
+        self.data = []
+        self.configure_and_load(config_element, tool_data_path, from_shed_config)
+
+    def configure_and_load(self, config_element, tool_data_path, from_shed_config=False, url_timeout=10):
+        self.rg_asset = config_element.get('asset', None)
+        assert self.rg_asset, ValueError('You must specify an asset attribute.')
+        super(RefgenieToolDataTable, self).configure_and_load(config_element, tool_data_path, from_shed_config=from_shed_config, url_timeout=url_timeout)
+
+    def parse_column_spec(self, config_element):
+        self.columns = {}
+        self.key_map = {}
+        self.template_for_column = {}
+        self.strip_for_column = {}
+        self.largest_index = 0
+        for i, elem in enumerate(config_element.findall('field')):
+            name = elem.get('name', None)
+            assert name, ValueError('You must provide a name refgenie field element.')
+            value = elem.text
+            self.key_map[name] = value
+            column_index = int(elem.get('column_index', i))
+
+            empty_field_value = elem.get('empty_field_value', None)
+            if empty_field_value is not None:
+                self.empty_field_values[name] = empty_field_value
+
+            self.template_for_column[name] = util.asbool(elem.get('template', False))
+            self.strip_for_column[name] = util.asbool(elem.get('strip', False))
+
+            self.columns[name] = column_index
+            self.largest_index = max(self.largest_index, column_index)
+        if 'name' not in self.columns:
+            self.columns['name'] = self.columns['value']
+
+    def parse_file_fields(self, filename, errors=None, here="__HERE__"):
+        rgc = refgenconf.RefGenConf(filename)
+        rval = []
+        for genome in rgc.list_genomes_by_asset(self.rg_asset):
+            genome_attributes = rgc.get_genome_attributes(genome)
+            description = genome_attributes.get('description', None)
+            asset_list = rgc.list(genome, include_tags=True)[genome]
+            for tagged_asset in asset_list:
+                asset, tag = tagged_asset.rsplit(':', 1)
+                if asset != self.rg_asset:
+                    continue
+                digest = rgc.id(genome, asset, tag=tag)
+                uuid = 'refgenie:%s/%s:%s@%s' % (genome, self.rg_asset, tag, digest)
+                display_name = description or '%s/%s' % (genome, tagged_asset)
+
+                def _seek_key(key):
+                    return rgc.get_asset(genome, asset, tag_name=tag, seek_key=key)
```

Comment:
If I may chime in:

Is there any particular reason you used `RefGenConf.get_asset` rather than `RefGenConf.seek` ?

`RefGenConf.get_asset` method is deprecated and will be removed in the next `refgenconf` release, see: https://github.com/databio/refgenconf/blob/0d9908decd2f880ca4928bfecad12b920ddf904e/refgenconf/refgenconf.py#L282-L313

We recommend using `RefGenConf.seek` instead. It both matches the CLI and does not assert asset existence _by default_. The latter is imperative when dealing with non-existing asset paths, like `hg38/bowtie2_index`, which returns a ready-to-use input for Bowtie2 software.

Reply:
he probably wrote that before we made that API change... but yes, should switch to `seek` now!

Reply:
No real reason, just missed swapping this one over during the update, I got the actual 'get_asset' one below it, will fix this one too. Nice catch, thanks.